### ISSUE TITLE
[BUG] resize of a disk in harvester_virtualmachine via terraform is returing success but does nothing

### ISSUE THREAD

**Issue by Martin-Weiss** [harvester/harvester#2810]
**Describe the bug**
Creation of a new VM with harvester_virtualmachine and then changing the size of a disk via terraform does not work as expected. Changing size via terraform apply succeeds but does nothing.

**To Reproduce**
Create a VM via terraform harvester_virtualmachine with two disks.
Then increase the size of the second disk.

**Expected behavior**
The size of the second disk should be increased.
In case this is not possible a failure should be reported.

**Environment**
 - Harvester ISO version: 1.0.3
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): bare metal

**Comment by w13915984028** [comment]
@futuretea Do you have new updates of this issue? thanks.

**Comment by futuretea** [comment]
I will try to handle disk resize in vm controller.

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.1.2-rc1`. Close this issue.

### Result
Verified can pass the following test cases 

#### Case 1: Resizing a volume via Volumes page doesn't break the VM. 
We can resize volume correctly from virtual machine edit page 

#### Case 2: Can't resize volumes on a running VM. 
  When we resize the volume size in a running VM on edit yaml, it will indicate error like
  ```
  please stop the VM before resizing volumes.
  ```
  ![image](https://user-images.githubusercontent.com/29251855/220115507-ae3cb2e6-172e-4c84-a0bd-ba540a9f9852.png)
  
  
  ![image](https://user-images.githubusercontent.com/29251855/220095542-8781c019-680f-4793-9f6b-33b857101954.png)

#### Case 3: Can't resize volumes to a smaller size. 
When we resize a volume to smaller size on edit yaml, it will indicate error like 
  ```
  PVC requests storage can't be less than previous value.
  ```
  ![image](https://user-images.githubusercontent.com/29251855/220095755-9b00a448-0bbe-4872-a38f-2b9f593a48fa.png)
  
  ![image](https://user-images.githubusercontent.com/29251855/220095946-eadb5992-1071-4e43-af16-eb277ff93f18.png)

#### Case 4: We can resize volumes through VM annotation. 
We can successfully resize volume to larger size after vm stopped in edit yaml 
  ![image](https://user-images.githubusercontent.com/29251855/220115654-5c2cc866-63dc-484c-b8ce-fa9b8daa47e5.png)
  
  ![image](https://user-images.githubusercontent.com/29251855/220115952-5d083d94-5039-4278-85b5-719d6fd067f9.png)


#### Case 5: terraform update volume size 
We can use terraform to update the volume size of stopped (Halted) VM.

![image](https://user-images.githubusercontent.com/29251855/220118688-ce1cb125-5a89-4f99-9e7e-eaca46f9ce55.png)

![image](https://user-images.githubusercontent.com/29251855/220118758-5e7d2fbc-9a1b-49ab-9bce-2489d98185ba.png)

### Test Information
* Test Environment: 1 node Harvester on local kvm machine
* Harvester version: `v1.1.2-rc1` 
* terraform provider version: v0.6.1

### Verify Steps
https://github.com/harvester/tests/issues/626#issue-1459257471

### Additional Context
* terraform resource.tf 
https://github.com/harvester/terraform-provider-harvester/blob/master/examples/resources/harvester_image/resource.tf
https://github.com/harvester/terraform-provider-harvester/blob/master/examples/resources/harvester_virtualmachine/resource.tf
